3] guessing game 2/main.py

```diff
@@ -0,0 +1,66 @@
+from random import randint
+import os
+
+min_number = 0
+max_number = 1000
+
+ai_number = randint(0, 1000)
+
+def mathematics():
+    return (min_number + max_number) // 2
+
+def compare(ai_number, user_answer):
+    global min_number, max_number
+
+    if user_answer == 1:
+        min_number = ai_number
+        return mathematics()
+
+    elif user_answer == 2:
+        max_number = ai_number
+        return mathematics()
+
+def clear():
+    os.system("clear")
+
+def lyar():
+    clear()  
+    input(f"Don't cheat!\nPress Enter to continue...")
+
+
+while True:
+    try:
+        clear()
+        user_number = int(input(f"Think about a number from 1 to 1000 and let me guess it!\nPlease enter your number: "))
+        if 1 <= user_number <= 1000:
+            break
+        else:
+            clear()
+            input(f"Please enter a number within the range of 1 to 1000.\nPress Enter to continue...")
+    except ValueError:
+        clear()
+        input(f"Please enter a valid integer.\nPress Enter to continue...")
+
+while True:
```

Comment:
velmi velmi velmi komplikovana logika. vubec nevim co se deje. ten algoritmus je pomerne primocary - je to klasicky binary search